src/components/LinkList.jsx

```diff
@@ -1,31 +1,50 @@
-import { useEffect, useState } from 'react';
-import { getUserInfo } from '../api/api';
+import { useCallback, useEffect, useState } from 'react';
+import { getUserInfo, getLinkInfo } from '../api/api';
 import LinkCard from './LinkCard';
 import styles from '../css/LinkList.module.css';
+import { useLocation } from "react-router-dom";
 
-function LinkList({ query }) {
+function LinkList({ query, id }) {
   const [folderData, setFolderData] = useState([]);
+  const { pathname } = useLocation();
+
+  const getLinkData = async (path, id) => {
+    const { data } = await getLinkInfo(path ,id);
+    
+    if (!data) return;
+
+    setFolderData(data);
+  }
 
   const getFolderData = async (path) => {
-    const result = await getUserInfo(path);
+    const { folder } = await getUserInfo(path);
     
-    if (!result) return;
+    if (!folder) return;
 
-    setFolderData(result.folder.links);
+    setFolderData(folder.links);
   }
 
+  const handleLoad = useCallback((path, id) => {
+    if (pathname === '/folder') {
+      getLinkData(path, id);
+      return;
+    }
+    if (pathname === '/shared') {
+      getFolderData(path, id)
+    }
+  }, [pathname]
+  )
+
   useEffect(() => {
-    getFolderData(query);
-  }, [query])
+    handleLoad(query, id);
+  }, [query, id, handleLoad])
 
   return (
     <div className={styles.content}>
       <div className={styles.items}>
-        {folderData.map((item) => {
-          return (
-            <LinkCard item={item} key={item.id}/>
-          )
-        })}
+        {folderData.length !== 0 ? (folderData.map((item) => 
```

Comment:
삼항연산자 쓰실때는 되도록 긍정조건으로 써주세요~
```suggestion
        {folderData.length ? 
```